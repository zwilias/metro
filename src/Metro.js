import MetroViewMode from './MetroViewMode';
import MetroViewRenderer from './MetroViewRenderer';
import MetroElementRenderer from './MetroElementRenderer';

const createElement = (elementType, props, ...children) => {
    props = props || {};
    children = children || [];

    if (elementType.prototype instanceof MetroViewMode) {
        return new MetroViewRenderer(elementType, props, children);
    }

    return new MetroElementRenderer(elementType, props, children);
}

const render = (element) => {
    element.beforeMount();
    return element.render();
}

const Metro = {
    render: render,
    createElement: createElement
};

export default Metro;
