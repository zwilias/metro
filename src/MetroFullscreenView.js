import MetroViewMode from './MetroViewMode';

class MetroFullscreenView extends MetroViewMode {
    constructor(props) {
        super(props);
    }

    renderComponent = (renderCallback) => {
        return new MAF.Class({
            id: 'something',
        	ClassName: 'MetroFullscreenView',
        	Extends: MAF.system.FullscreenView,

        	// Create your view template
        	createView: function () {
                var view = this;
                renderCallback(view);
        	}
        });
    }
}

export default MetroFullscreenView;
