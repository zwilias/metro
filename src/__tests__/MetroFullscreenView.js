import MetroFullscreenView from '../MetroFullscreenView';
import Metro from '../Metro';
jest.disableAutomock();

describe('MetroFullscreenView', () => {
    describe('constructor', () => {
        it('should set props', () => {
            const props = {};
            const result = new MetroFullscreenView(props);

            expect(result.props).toBe(props);
        });
    });

    describe('e2e', () => {
        it('should return a MAF.Class', () => {
            const mafMock = (<MetroFullscreenView />).renderContainer();
            expect(mafMock.settings).toMatchSnapshot();
        });
    });
});
