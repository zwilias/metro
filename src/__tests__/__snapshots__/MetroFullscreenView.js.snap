exports[`MetroFullscreenView renderComponent should return a MAF.Class 1`] = `"{\"id\":\"something\",\"ClassName\":\"MetroFullscreenView\",\"Extends\":\"FullscreenView\"}"`;
